lib/parse.js

```diff
@@ -5,7 +5,9 @@ const languages = require('./languages');
 
 const constants = {
   ESCAPED_CHAR_REGEX: /^\\./,
-  QUOTED_STRING_REGEX:/^(['"`])((?:|[^\1])+?)(\1)/,
+  QUOTED_STRING_REGEX_1: /^"[^"\\]*(?:\\[\s\S][^"\\]*)*"/,
+  QUOTED_STRING_REGEX_2: /^'[^'\\]*(?:\\[\s\S][^'\\]*)*'/,
+  QUOTED_STRING_REGEX_3: /^`[^`\\]*(?:\\[\s\S][^`\\]*)*`/,
```

Comment:
Is there a reason why `(['"`])` is expanded into 3 regexes?

Reply:
Because I don't think it's possible to reference a group inside a negated set. The previous implementation seemed to do that, but it didn't seem to work.

Reply:
Happy to be proven wrong if that is the case - I am not an Regex expert 😅

Reply:
ah, you are correct!